articles/150_roslaunch.md

```diff
@@ -0,0 +1,562 @@
+---
+layout: default
+title: ROS 2 Launch System
+permalink: articles/roslaunch.html
+abstract:
+  The launch system in ROS is responsible for helping the user describe the configuration of their system and then execute it as described. The configuration of the system includes what programs to run, what arguments to pass them, and ROS specific conventions which make it easy to reuse components throughout the system by giving them each different configurations. Also, because the launch system is the process (or the set of processes) which executes the user's processes, it is responsible for monitoring the state of the processes it launched, as well as reporting and/or reacting to changes in the state of those processes.
```

Comment:
I hate to make a complicated subject even more complicated, but what do we think about executing processes across multiple different compute modules?  After all, a reasonably complex robot will have multiple processing modules, and hence successful launching of the "system" really involves launching processes on all of the modules.  There's also the related, even larger question about launching swarms of robots, but that gets a little pie-in-the-sky.

If it is not our intention to handle either of those cases here, I think that is fine, but maybe we want to explicitly say that it is a non-goal of this, and larger orchestration may come in a later design document?

Reply:
I think it is a goal of the overall system and therefore this design doc (you've noticed that it's mentioned below), but not a feature we have to implement immediately if time doesn't allow, so long as we design for it.

Reply:
At the moment (i.e., roslaunch 1) the system does several things
 - what to run
   - including launch-prefixes
 - how to run (largely implicitly, but worth mentioning)
 - placement (where to run)
   - including remote-node access 
 - configuration
   - parameters (in several different ways)
   - command-line arguments
   - env-handlers
 - data connectivity (remap, group)
 - error handling (respawn, required)
 - test definition

Arguably, this was not really necessary, but just came about because it was a convenient place to stick all the stuff.

This definitely caused usability issues already in ROS1. I  have rewritten countless launch files, instead of just including them, because one of these things was defined in a way that did not match requirements.

As you eloquently discuss in the remainder of  the document, each of these issues becomes more complex with ROS2.

Thus, we might want to revisit whether we can break this up, and create a combination of simpler tools.